tests/e2e/test_triton.py

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+from pydsl.frontend import compile
+from pydsl.type import F32, UInt8, UInt32, Bool, Index, Number
+from pydsl.memref import MemRefFactory, DYNAMIC, alloc, MemRef
+from pydsl.affine import affine_range as arange, symbol as S
+from helper import run
+import pytest
+
+skip = False
+try:
+    import torch
+    import triton
+    import triton.language as tl
+except ImportError:
+    skip = True
+
+hastriton = pytest.mark.skipif(
+    skip, reason="Triton or PyTorch is not installed"
+)
+
+
+@hastriton
```

Comment:
```suggestion

try:
    import torch
    import triton
    import triton.language as tl
    has_triton = True
except ImportError:
    has_triton = False

pytestmark = pytest.mark.skipif(
    not has_triton, reason="Triton or PyTorch is not installed"
)


```

This skips all functions in the module see https://docs.pytest.org/en/stable/how-to/skipping.html#skip-all-test-functions-of-a-class-or-module.

It's also possible to skip an entire file or directory (so even code outside of test functions isn't run) https://docs.pytest.org/en/stable/example/pythoncollection.html#customizing-test-collection, but I don't think we need that.